### PR TITLE
Coverage to 90%

### PR DESCRIPTION
**Pull request by mavaylon1**
## Motivation

Getting coverage to at least 90% and going through deprecations for HDMF 4.0

## Checklist

- [x] Did you update `CHANGELOG.md` with your changes?
- [x] Does the PR clearly describe the problem and the solution?
- [x] Have you reviewed our [Contributing Guide](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst)?
- [ ] Does the PR use "Fix #XXX" notation to tell GitHub to close the relevant issue numbered XXX when the PR is merged?

### CODE REVIEW

File: src/hdmf/backends/hdf5/h5tools.py [hunk 1]
```diff
@@ -1570,17 +1517,6 @@ def __queue_ref(self, func):
         # dependency
         self.__ref_queue.append(func)
 
-    def __rec_get_ref(self, ref_list):
```
Comment (mavaylon1) [inline src/hdmf/backends/hdf5/h5tools.py:1573]: There is not partial coverage of this function. This feels like it should be just a static method if you guys actually want to keep this.

**Comment by mavaylon1** [comment]
Review Notes: I did a few rounds of searches to make sure anything region reference related is removed. There is one method I left a note on. HDMF_Zarr needs to be updated when this is merged.

File: src/hdmf/backends/hdf5/h5_utils.py [hunk 2]
```diff
@@ -85,7 +83,7 @@ def append(self, dataset, data):
 
 
 class H5Dataset(HDMFDataset):
-    @docval({'name': 'dataset', 'type': (Dataset, Array), 'doc': 'the HDF5 file lazily evaluate'},
+    @docval({'name': 'dataset', 'type': (Dataset), 'doc': 'the HDF5 file lazily evaluate'},
```
Comment (rly) [inline src/hdmf/backends/hdf5/h5_utils.py]:
```suggestion
    @docval({'name': 'dataset', 'type': Dataset, 'doc': 'the HDF5 file lazily evaluate'},
```

File: src/hdmf/backends/hdf5/h5_utils.py [hunk 3]
```diff
@@ -189,7 +183,7 @@ def get_object(self, h5obj):
 
 class AbstractH5TableDataset(DatasetOfReferences):
 
-    @docval({'name': 'dataset', 'type': (Dataset, Array), 'doc': 'the HDF5 file lazily evaluate'},
+    @docval({'name': 'dataset', 'type': (Dataset), 'doc': 'the HDF5 file lazily evaluate'},
```
Comment (rly) [inline src/hdmf/backends/hdf5/h5_utils.py]:
```suggestion
    @docval({'name': 'dataset', 'type': Dataset, 'doc': 'the HDF5 file lazily evaluate'},
```

File: src/hdmf/backends/hdf5/h5tools.py [hunk 4]
```diff
@@ -1524,21 +1475,10 @@ def __get_ref(self, **kwargs):
...
-        if isinstance(container, RegionBuilder):
-            region = container.region
-        if region is not None:
-            dset = self.__file[path]
-            if not isinstance(dset, Dataset):
-                raise ValueError('cannot create region reference without Dataset')
-            return self.__file[path].regionref[region]
-        else:
-            return self.__file[path].ref
+        return self.__file[path].ref
```
Comment (rly) [inline src/hdmf/backends/hdf5/h5tools.py:1479]:
```suggestion
        self.logger.debug("Getting reference at path '%s'" % path)
        return self.__file[path].ref
```

File: src/hdmf/backends/hdf5/h5tools.py (hunk 1 above)
Comment (rly) [reply src/hdmf/backends/hdf5/h5tools.py:1573]: Where is this used? My quick search of the codebase suggests it is not used at all. Why was it added?

File: src/hdmf/query.py [hunk 5]
```diff
@@ -2,96 +2,9 @@
...
-    def __xor__(self, other):
-        return NotImplemented
-
-    def __contains__(self, other):
-        return NotImplemented
-
-
 @docval_macro('array_data')
 class HDMFDataset(metaclass=ExtenderMeta):
     __operations__ = (
```
Comment (rly) [inline src/hdmf/query.py]: `__operations__` can be cut too

File: src/hdmf/query.py [hunk 6]
```diff
@@ -103,42 +16,19 @@ class HDMFDataset(metaclass=ExtenderMeta):
...
-            if isinstance(key, Query):
-                return key.evaluate()
             return key
 
     def __getitem__(self, key):
         idx = self.__evaluate_key(key)
         return self.dataset[idx]
 
-    @docval({'name': 'dataset', 'type': ('array_data', Array), 'doc': 'the HDF5 file lazily evaluate'})
+    @docval({'name': 'dataset', 'type': ('array_data'), 'doc': 'the HDF5 file lazily evaluate'})
```
Comment (rly) [inline src/hdmf/query.py]:
```suggestion
    @docval({'name': 'dataset', 'type': 'array_data', 'doc': 'the HDF5 file lazily evaluate'})
```

**Comment by rly** [comment]
This is great. Could you also remove `fmt_docval_args` from `hdmf/utils.py`?

**Comment by rly** [comment]
And `call_docval_func` and the blocks of commented functions in `hdmf/utils.py`

File: src/hdmf/backends/hdf5/h5tools.py (hunk 1 above)
Comment (rly) [reply src/hdmf/backends/hdf5/h5tools.py:1573]: If it's not used and not valuable, let's remove it.

Comment (mavaylon1) [reply src/hdmf/backends/hdf5/h5tools.py:1573]: I have no idea. I also saw that it was not used anywhere. A git blame shows it was made by Andrew 4 years ago. @rly

**Review by rly — APPROVED** [review]

**Pull request merged**